### PR TITLE
[T2] appconfiguration

### PR DESCRIPTION
**Pull request by 00Kai0**
When I try to test appconfiguration, I got two question:
1. miss `privateEndpointConnections` in the response of configuration store. Then response example:
```json
{
  "type":"Microsoft.AppConfiguration/configurationStores",
  "properties":{
    "provisioningState":"Succeeded",
    "creationDate":"2020-03-25T04:21:41+00:00",
    "endpoint":"https://configuration39c8158a.azconfig.io",
    "encryption":{"keyVaultProperties":null},
    "privateEndpointConnections":[
      {
        "id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test_cli_mgmt_appconfiguration_test_appconfiguration39c8158a/providers/Microsoft.AppConfiguration/configurationStores/configuration39c8158a/privateEndpointConnections/endpointxyz.f54b944c-7053-40e1-9791-5b8297d35d18",
        "name":"endpointxyz.f54b944c-7053-40e1-9791-5b8297d35d18",
        "type":"Microsoft.AppConfiguration/configurationStores/privateEndpointConnections",
        "properties":{
          "provisioningState":"Succeeded",
          "privateEndpoint":{
            "id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test_cli_mgmt_appconfiguration_test_appconfiguration39c8158a/providers/Microsoft.Network/privateEndpoints/endpointxyz"
          },
          "privateLinkServiceConnectionState":{
            "status":"Approved",
            "description":"Auto-Approved",
            "actionsRequired":"None"
          }
        }
      }
    ]
  },
  "sku":{"name":"standard"},
  "id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test_cli_mgmt_appconfiguration_test_appconfiguration39c8158a/providers/Microsoft.AppConfiguration/configurationStores/configuration39c8158a",
  "name":"configuration39c8158a",
  "location":"westus",
  "tags":{"my_tag":"myTagValue"}
}
```

2. when I try to create or update private endpoint connection, I always get an error:
```
azure.core.exceptions.HttpResponseError: (InvalidProperty) Some of the properties of 'PrivateEndpointConnection' are invalid. Errors: 'Missing required property 'Id'.'
```
So i think this `id` can not be ignore in request.
And also I want to find out whether `appconfiguration` can create private endpoint by itself. 
In other services, private endpoint is created by `network`

### CODE REVIEW

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2019-11-01-preview/appconfiguration.json [hunk 1]
```diff
@@ -1374,8 +1382,7 @@
       "properties": {
         "id": {
           "description": "The resource ID.",
-          "type": "string",
-          "readOnly": true
```
Comment (raych1) [inline specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2019-11-01-preview/appconfiguration.json]: It's breaking change for removing readonly.

**Comment by majastrz** [comment]
Please address Ray's breaking change comment. Otherwise, changes look good from ARM side.

**Comment by 00Kai0** [comment]
Don't merge

**Comment by raych1** [comment]
@00Kai0 , can you please resolve the branch conflicts and the comments?

**Comment by 00Kai0** [comment]
@raych1 sorry, too late. This issue seems to have been fixed by another one. #9002
it could be closed, thx.

**Pull request closed without merging**